### PR TITLE
🚀Release To Preview - v1.0.0-preview.8

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Release Pull Request</h1>

<h2 style="font-weight:bold" align="center">✅Development Checklist✅</h2>

<details open><summary>🌳Branching🌳</summary>

Does the name of the head(source) branch for this pull request have the correct **_preview branch_** syntax?

Syntax: _preview/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;-preview.&lt;prev-number&gt;_  
Example: _preview/v1.2.3-preview.4_
- [x] Yes
- [ ] No

Does the name of the base(target) branch for this pull request have the correct **_release branch_** syntax?

Syntax: _release/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;_  
Example: _release/v1.2.3_
- [x] Yes
- [ ] No

Is the head(source) branch for this pull request created from a branch with the correct **_release branch_** syntax?

Syntax: _release/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;_  
Example: _release/v1.2.3_
- [x] Yes
- [ ] No
</details>

<details open><summary>🐛Bugs🐛</summary>

Contains Bug Fix(es)
- [x] Yes
  - [x] A bug-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>💣Breaking Change(s)💣</summary>

Contains [breaking change(s)](https://docs.microsoft.com/en-us/dotnet/core/compatibility/#modifications-to-the-public-contract)
- [x] Yes
  - [x] A breaking-change-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>✨Enhancements✨</summary>

Contains enhancement(s) that add a feature or behavior.
- [x] Yes
  - [x] An enhancement-label label has been added to the pull request.
- [ ]  No
</details>

<details open><summary>⚙️Workflow (CI/CD) Changes⚙️</summary>

Contains changes to workflow files. These changes can only done by the project maintainers.
- [x] Yes
  - [x] A workflow-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>📃Documentation Updates📃</summary>

Contains changes that require updates to existing **_code_** and/or **_Velaptor product_** documentation.
- [ ] Yes
  - [ ] I have updated the documentation accordingly.
    - Choose the type of documentation.
    - [ ] Code Documentation
      - [ ] A documentation-code-label label has been added to the pull request.
    - [ ] Product Documentation
      - [ ] A documentation-product-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>🧪Manual Testing🧪</summary>

I have manually tested my changes. (This can be done by using project named **_VelaptorTesting_**).
- [ ] Yes
- [x] No
</details>

## Issues Released:
- #221
- #204
- #199
- #197
- #196
- #192
- #189
- #110
- #94
- #71

<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] Pull request title matches the example below with the correct version.
  - Example: 🚀Release To Preview - v1.2.3-preview.4
- [x] Issues exist and are linked to this pull request.
- [x] My code follows the coding style of this project.
  - The style is enforced by the *.editorconfig* files in the project and displayed as warnings.  If there is an edge case with a coding style that should be ignored or changed, reach out, and let's discuss it.
- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vise-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - This does not mean that the tests pass in both the local and CI environment.
  - Status checks run for this pull request when merging into the **_release/v\*.\*.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.
- [x] Update library version by updating the **_\<Version/\>_** and **_\<FileVersion/\>_** tags in the **_Velaptor_** **_.csproj_** file.
  - Every change to a pull request will run a status check to confirm that the version has the correct syntax, a tag does not exist, and that it has not already been published to [nuget](https://www.nuget.org/)
  - Make sure to add the **_-preview.\<number\>_** syntax to the end of the version.
    Example:
    ``` html
    <Version>1.2.3-preview.4</Version>
    <FileVersion>1.2.3-preview.4</FileVersion>
    ```
- [ ] I have updated the release notes by creating a new preview release notes file and adding it to the **_./Documentation/ReleaseNotes/PreviewReleases_** folder.

### CODE REVIEW

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+As with all software, including releases, there is always a chance for issues and bugs.  It is also common to miss changes in the release notes when there are many.  This is even more common in preview releases.
+
+---
+
+<h2 style="font-weight:bold" align="center">New 🎉</h2>
+
+1. Added new constructor to `CornerRadius` struct for convenience.  
+   GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+    * This make it easier to set all of the radius values to the same value.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md:13]: Spelling
```suggestion
    * This makes it easier to set all of the radius values to the same value.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    * This will automatically update the `Position` property of the rectangle.
+5. Added a new property named `Right` to the `RectShape` struct to get or set the location of the top of the rectangle.  
+   GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+    * This will automatically update the `Position` property of the rectangle.
+6. Added a new property named `Bottom` to the `RectShape` struct to get or set the location of the top of the rectangle.  
+   GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+    * This will automatically update the `Position` property of the rectangle.
+1.  Added a new property named `Left` to the `RectShape` struct to get or set the location of the top of the rectangle.  
+    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+    * This will automatically update the `Position` property of the rectangle.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md:29]: Location of properties is incorrect in the text.  Make sure line numbering is sequential throughout the whole file.
```suggestion
5. Added a new property named `Right` to the `RectShape` struct to get or set the location of the right of the rectangle.  
   GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
    * This will automatically update the `Position` property of the rectangle.
6. Added a new property named `Bottom` to the `RectShape` struct to get or set the location of the bottom of the rectangle.  
   GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
    * This will automatically update the `Position` property of the rectangle.
1.  Added a new property named `Left` to the `RectShape` struct to get or set the location of the left of the rectangle.  
    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
    * This will automatically update the `Position` property of the rectangle.
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+1.  Added new `System.Drawing.Color` extension method to increase the brightness of the color.  
+    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+9.  Added new `System.Drawing.Color` extension method to decrease the brightness of the color.  
+    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+10. Changed the `ControlBase.Width` and `ControlBase.Height` property setters from `protected internal` to `public`  
+    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+11. Added new property named `AutoSize` to the `Label` control.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md:36]: Wordy
```suggestion
11. Added property named `AutoSize` to the `Label` control.  
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+5. Added a new property named `Right` to the `RectShape` struct to get or set the location of the top of the rectangle.  
+   GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+    * This will automatically update the `Position` property of the rectangle.
+6. Added a new property named `Bottom` to the `RectShape` struct to get or set the location of the top of the rectangle.  
+   GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+    * This will automatically update the `Position` property of the rectangle.
+1.  Added a new property named `Left` to the `RectShape` struct to get or set the location of the top of the rectangle.  
+    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+    * This will automatically update the `Position` property of the rectangle.
+1.  Added new `System.Drawing.Color` extension method to increase the brightness of the color.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md:30]: Call out names of the methods

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+12. Added new property named `AutoSize` to the `Button` control.  
+    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+    * When set to `true`, the button's `Width` and `Height` will automatically be set based on the width and height of the button text.  When set to `false`, `Width` and `Height` are independent of the button text size.
+13. Added new method to the `IFont` interface and `Font` class that returns the bounds of each character of some given text relative to a position.  
+    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+14. Added the ability to add a `Label` to a `Button`.  This can be done by sending in a `Label` instance into the `Button` constructor.  
+    GitHub Issues: [#196](https://github.com/KinsonDigital/Velaptor/issues/196)
+15. Added added additional constructors to the `Button` class to improve the ease of use when creating new buttons.  List of new constructors listed below.
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md:46]: Wordy
```suggestion
15. Added additional constructors to the `Button` class to improve the ease of use when creating new buttons.  List of new constructors listed below.  
```

File: Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+---
+
+<h2 style="font-weight:bold" align="center">Breaking Changes 💣</h2>
+
+1. Changed the order of rendering from textures, text, then rectangle primitives to textures, rectangle primitives, then text.
+2. Changed the behavior of the `RectShape.BorderThickness` and `RectShape.CornerRadius` properties
+    * Previously, both properties would check the incoming value and restrict them to a limit of the smallest value between the half width or half height.  This was done to prevent any unintended rendering artifacts when rendering rectangles.  This behavior was removed from the `RectShape` and moved to the internal `RectGPUBuffer` class.  These checks and restrictions now occur right before sending the data to the GPU.
+3. Changed the `ControlBase.IsLoaded` property from `protected` to `private`
+    * The property was meant to be updated via the `ControlBase.LoadContent()` method
```
Comment (kselena) [inline Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.8.md:75]: Punctuation (add periods at end of sentences).  Check full document there are more punctuation issues.
```suggestion
3. Changed the `ControlBase.IsLoaded` property from `protected` to `private`.
    * The property was meant to be updated via the `ControlBase.LoadContent()` method.
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by kselena — APPROVED** [review]
